ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -532,7 +534,8 @@ public static long getTtlFromRequestHeader(Map<String, Object> args) throws Rest
    * @return the user metadata extracted from arguments.
    * @throws RestServiceException if usermetadata arguments have null values.
    */
-  public static byte[] buildUserMetadata(Map<String, Object> args) throws RestServiceException {
+  public static byte[] buildUserMetadata(Map<String, Object> args)
+      throws RestServiceException, UnsupportedEncodingException {
```

Comment:
can we convert the UnsupportedEncodingException to RestServiceException so we don't have to change the method signature?

Reply:
Sure, updated.